
Using device: cpu

Epoch[5/5], Loss: 5062.2489: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:03<00:00,  1.31it/s]
Traceback (most recent call last):
  File "C:\Users\tobia\Documents\GitHub\RNN_Tobia_Simmler\main.py", line 15, in <module>
    run(config["experiment"])
  File "C:\Users\tobia\Documents\GitHub\RNN_Tobia_Simmler\src\runner.py", line 66, in run
    plot_h(tm, config["options"])
  File "C:\Users\tobia\Documents\GitHub\RNN_Tobia_Simmler\src\vizual.py", line 63, in plot_h
    custom_plot(possible_move_vals, pred_move_val_counter, total_move_val_counter, path, params,  "move.pdf")
  File "C:\Users\tobia\Documents\GitHub\RNN_Tobia_Simmler\src\vizual.py", line 36, in custom_plot
    wandb.log({name: fig})
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 394, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 345, in wrapper_fn
    return func(self, *args, **kwargs)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 335, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 1751, in log
    self._log(data=data, step=step, commit=commit)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 1532, in _log
    self._partial_history_callback(data, step, commit)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\wandb_run.py", line 1402, in _partial_history_callback
    self._backend.interface.publish_partial_history(
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\sdk\interface\interface.py", line 579, in publish_partial_history
    item.value_json = json_dumps_safer_history(v)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\util.py", line 782, in json_dumps_safer_history
    return json.dumps(obj, cls=WandBHistoryJSONEncoder, **kwargs)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\site-packages\wandb\util.py", line 747, in default
    return json.JSONEncoder.default(self, obj)
  File "C:\Users\tobia\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Line2D is not JSON serializable
--------------- Train ---------------
0 	 0.0176 	 0.0
100 	 -0.0361 	 0.0
200 	 -0.0361 	 0.0
300 	 -0.0361 	 0.0
400 	 -0.0361 	 0.0
500 	 -0.0361 	 0.0
600 	 -0.0361 	 0.0
700 	 0.3712 	 -0.0028
800 	 0.1207 	 -0.1512
900 	 0.0197 	 -0.2903
1000 	 -0.1284 	 -0.3007
1100 	 0.269 	 -0.2925
1200 	 0.0081 	 -0.3573
1300 	 -0.1571 	 -0.5023
-------------------------------------
--------------- Example Test ---------------
0 	 0.1718 	 0.0
100 	 0.118 	 0.0
200 	 0.118 	 0.0
300 	 0.118 	 0.0
400 	 0.118 	 0.0
500 	 0.118 	 0.0
600 	 0.118 	 0.0
700 	 0.0914 	 -0.1919
800 	 0.1293 	 -0.5212
900 	 0.1224 	 -0.8281
1000 	 0.1726 	 -1.0
1100 	 0.1945 	 -1.0
1200 	 0.0416 	 -1.0
1300 	 0.132 	 -1.0
-------------------------------------
Model last
{'test_loss': [4219.7666015625]}